ApiLocalStorageVideoRecorder/Pages/Players/PlayersViewModel.swift

```diff
@@ -0,0 +1,38 @@
+//
+//  PlayersViewModel.swift
+//  ApiLocalStorageVideoRecorder
+//
+//  Created by Çağrı Tuğberk Masat on 1.06.2023.
+//
+
+import Foundation
+
+protocol PlayersViewModelDelegate: AnyObject {
+    func didFetchPlayers()
+}
+
+class PlayersViewModel {
+    private let coreDataManager: CoreDataManager
+    private var players: [Player] = []
+    weak var delegate: PlayersViewModelDelegate?
+    
+    init(coreDataManager: CoreDataManager) {
+        self.coreDataManager = coreDataManager
+    }
+    
+    func fetchPlayers() {
+        players = coreDataManager.fetchPlayers()
+        delegate?.didFetchPlayers()
+    }
+    
+    func numberOfPlayers() -> Int {
+        return players.count
+    }
+    
+    func getPlayer(at index: Int) -> Player? {
```

Comment:
Model name of Player should be changed as PlayerModel also this change should be done for Shot. Check it later.